src/plstring.c

```diff
@@ -0,0 +1,540 @@
+        const u16 * charMap = (charWidth == CHARACTER_WIDTH_FULL) ? FULL_WIDTH_DIGITS : HALF_WIDTH_DIGITS;
+
+        PLString_Clear(string);
+
+        // curse forwards through the integer digit by digit, appending individual digits along the way
+        for (power = POWERS_OF_TEN[len - 1]; power > 0; power /= 10) {
+            digit     = (u16) (number / power);
+            remainder = (u64) (number - (power * digit));
+
+            if (paddingMode == NUMBER_PADDING_MODE_ZEROES)      // this branch is hit if the user specified this padding mode, or any time after we hit a non-zero digit
+            {
+                string->buffer[string->len++] = (digit < 10) ? charMap[digit] : 0xe2;
+            }
+            else if (digit != 0 || power == 1)                  // once we hit this branch, we should always hit the above branch on future digits
+            {
+                paddingMode = NUMBER_PADDING_MODE_ZEROES;
+                string->buffer[string->len++] = (digit < 10) ? charMap[digit] : 0xe2;
+            }
+            else if (paddingMode == NUMBER_PADDING_MODE_SPACES) // not in zero-padding mode, and found a leading zero, pad out with the appropriate width space
+            {
+                string->buffer[string->len++] = (charWidth == CHARACTER_WIDTH_FULL) ? 0x1 : 0x1e2;
+            }
+
+            number = remainder;
+        }
+
+        string->buffer[string->len] = TERMINATOR;
+        return;
+    }
+
+    GF_ASSERT(0);
+}
+
+u64 PLString_ToInt64 (const PLString * src, BOOL * successful)
+{
+    u64 digit = 0, result = 0, power = 1;
+    int i;
+    PLChar c;
+    u16 zeroFullWidth = 0xa2;
+    u16 zeroHalfWidth = 0x121;
+
+    // We don't support integers longer than 18 digits
+    if (string->len > 18) {
+        return 0;
+    }
+
+    // work through the string backwards, starting from the 1's place
+    for (i = (string->len - 1); i >= 0; i--, power *= 10) {
+        c = string->buffer[i];
+
+        // check if this is a digit on the full-width map
+        digit = c - zeroFullWidth;
+        if (digit >= 10)
+        {
+            // check if this is a digit on the half-width map
+            digit = c - zeroHalfWidth;
+            if (digit >= 10)
+            {
+                // fail, but return as much of the number as we could parse
+                *successful = 0;
+                return result;
+            }
+        }
+
+        // slot the digit into the result in the appropriate place, then go up to the next place
+        digit = digit * power;
+        result += digit;
+    }
+
+    // success!
+    *successful = 1;
+    return result;
+}
+
+int PLString_Compare (const PLString * this, const PLString * that)
+{
+    PLSTRING_CHECK(this);
+    PLSTRING_CHECK(that);
+
+    for (int i = 0; this->buffer[i] == that->buffer[i]; i++) {
+        // Both strings matched on a terminator; the strings match
+        if (this->buffer[i] == TERMINATOR) {
+            return 0;
```

Comment:
`FALSE`

Reply:
I'll fight on these two in particular, because `return FALSE` doesn't make sense in the context of the function for this particular branch. Here, the routine is asking for a comparison between the two functions, so returning `FALSE` is a negation of expectations. Something we could do stylistically is change the name of the function to `_Differ` or something like that.

Reply:
You're right I forgot how strcmp works.